set.js

```diff
@@ -0,0 +1,90 @@
+class Set {
+  constructor(array) {
+    this.values = array || [];
+  }
+  add(value) {
+    this.values.push(value);
+  }
+  remove(value) {
+    this.values.splice(this.values.indexOf(value), 1);
```

Comment:
if you try to remove a value that doesn't exist in the set `indexOf` will return -1. That will make splice remove the last item.